net/usbwlan.c

```diff
@@ -0,0 +1,521 @@
+/*
+ * Phoenix-RTOS
+ *
+ * USB WLAN driver
+ *
+ * Copyright 2025 Phoenix Systems
+ * Author: Julian Uziembło
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/minmax.h>
+#include <sys/threads.h>
+#include <unistd.h>
+
+#include <posix/utils.h>
+
+#include <usbdriver.h>
+#include <usbhost.h>
+
+#include "sys/msg.h"
+#include "usb.h"
+
+#include "usbwlan.h"
+
+#define USBWLAN_N_MSGTHREADS   2
+#define USBWLAN_MSGTHREAD_PRIO 2
+
+
+/* singular device */
+typedef struct {
+	idnode_t node;
+	usb_devinfo_t dev;
+
+	/* USB HOST */
+	int pipeCtrl;
+	int pipeIntIN;
+	int pipeBulkIN;
+	int pipeBulkOUT;
+
+	char path[32];
+
+	int flags;
+	pid_t clientpid;
+	volatile int refcnt; /* protected by usbwlan_common.lock */
+
+	usb_driver_t *drv;
+} usbwlan_dev_t;
+
+
+static struct {
+	idtree_t devices;
+	unsigned msgport;
+	handle_t lock;
+	char msgthread_stacks[USBWLAN_N_MSGTHREADS][1024] __attribute__((aligned(8)));
+} usbwlan_common;
+
+
+static const usb_device_id_t supported_devices[] = {
+	/* Sterling LWB5+ USB Dongle */
+	{ 0x04b4, 0x0bdc, USBDRV_ANY, USBDRV_ANY, USBDRV_ANY },
+	{ 0x04b4, 0xbd29, USBDRV_ANY, USBDRV_ANY, USBDRV_ANY }
+};
+
+
+static usbwlan_dev_t *_usbwlan_devAlloc(void)
+{
+	usbwlan_dev_t *dev;
+
+	dev = calloc(1, sizeof(*dev));
+	if (dev == NULL) {
+		fprintf(stderr, "usbwlan: Not enough memory for new device\n");
```

Comment:
I personally prefer to introduce `log_{info,warning,error,debug}` macros in new code as this makes the strings shorter and makes changes to the logging provider easier, but this is up to you